### PR TITLE
refactor(sqlite): add possible failure handling around sqlite multithreading

### PR DESCRIPTION
**Pull request by benmarten**
Fixes possible sporadic/rare multithreading issues around sqlite db.

## Proposed Changes
  - Set Sqlite config to use serialized mode.
  - Open DB connection using full mutex/serialized mode.
  - Adds try/catch around method implementation

### CODE REVIEW

File: Leanplum-SDK/Classes/LPDatabase.m [hunk 1]
```diff
@@ -49,8 +49,17 @@ - (id)init
  */
 - (sqlite3 *)initSQLite
 {
+    if (sqlite3_threadsafe() != SQLITE_CONFIG_SERIALIZED) {
```
Comment (alexisoyama) [inline Leanplum-SDK/Classes/LPDatabase.m]: As discussed offline, let's should remove this block.

Note from https://sqlite.org/c3ref/initialize.html
"The sqlite3_initialize() interface is threadsafe, but sqlite3_shutdown() is not. The sqlite3_shutdown() interface must only be called from a single thread. All open database connections must be closed and all other SQLite resources must be deallocated prior to invoking sqlite3_shutdown()."

There are no guarantee that other SDK won't call this.

Comment (benmarten) [reply Leanplum-SDK/Classes/LPDatabase.m]: yep. removed it. good catch too :)

File: Leanplum-SDK/Classes/LPDatabase.m [hunk 2]
```diff
@@ -50,7 +50,8 @@ - (id)init
 - (sqlite3 *)initSQLite
 {
     const char *sqliteFilePath = [[LPDatabase sqliteFilePath] UTF8String];
-    int result = sqlite3_open(sqliteFilePath, &sqlite);
+    int result = sqlite3_open_v2(sqliteFilePath, &sqlite, SQLITE_OPEN_CREATE | SQLITE_OPEN_READWRITE | SQLITE_OPEN_FULLMUTEX, NULL);
+
```
Comment (alexisoyama) [inline Leanplum-SDK/Classes/LPDatabase.m]: line

**Comment by benmarten** [comment]
test this please

**Comment by benmarten** [comment]
test this please

**Pull request merged**